clients/da-vinci-client/src/test/java/com/linkedin/davinci/replication/merge/TestMergeUpdateWithFieldLevelTimestamp.java

```diff
@@ -1,21 +1,334 @@
 package com.linkedin.davinci.replication.merge;
 
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.ACTIVE_ELEM_TS_FIELD_NAME;
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.DELETED_ELEM_FIELD_NAME;
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.DELETED_ELEM_TS_FIELD_NAME;
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.PUT_ONLY_PART_LENGTH_FIELD_NAME;
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.TOP_LEVEL_COLO_ID_FIELD_NAME;
+import static com.linkedin.venice.schema.rmd.v1.CollectionRmdTimestamp.TOP_LEVEL_TS_FIELD_NAME;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import com.linkedin.davinci.replication.RmdWithValueSchemaId;
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.SchemaUtils;
+import com.linkedin.venice.schema.rmd.RmdConstants;
+import com.linkedin.venice.schema.writecompute.DerivedSchemaEntry;
+import com.linkedin.venice.schema.writecompute.WriteComputeSchemaConverter;
+import com.linkedin.venice.serializer.avro.MapOrderingPreservingSerDeFactory;
+import com.linkedin.venice.utils.lazy.Lazy;
+import com.linkedin.venice.writer.update.UpdateBuilder;
+import com.linkedin.venice.writer.update.UpdateBuilderImpl;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.util.Utf8;
+import org.testng.Assert;
 import org.testng.annotations.Test;
 
 
+/**
+ * TODO: Use schema that contains map field after Avro Map deserialization issue is resolved.
+ * TODO: Merge with {@link TestMergeUpdateWithValueLevelTimestamp}.
+ */
 public class TestMergeUpdateWithFieldLevelTimestamp extends TestMergeConflictResolver {
   @Test
-  public void testUpdateIgnored() {
-    // TODO
+  public void testUpdateIgnoredFieldUpdate() {
+    final int incomingValueSchemaId = 3;
+    final int incomingWriteComputeSchemaId = 3;
+    final int oldValueSchemaId = 3;
+    // Set up
```

Comment:
Thanks for the review. I have been thinking about this - so some chunks of the logics in this test have great overlap with the other one over ValueLevelTimeStamp (if a value was put first then it will have value level TS, and follow up update will convert it into field level TS). I am thinking after I merge the other PR I might want to rewrite this PR to reduce the duplications.

Reply:
Sure. It's usually better to do it altogether otherwise we will forget it easily. 
Let's resolve the conflicts and get this merged.